### DISCUSSION TITLE
If I want to calculate the coupled stiffness matrix K_uT or K_Tu for a two-field coupled problem?

### DISCUSSION THREAD

**Discussion by zyhsjtu** [Ferrite-FEM/Ferrite.jl#1111] [Q&A]
For a two-field problem,i.e., thermo-elastic problem. If I want to calculate the global coupling matrices K_uT or K_Tu, is there a way to achieve this efficiently? If I define Kud as ` K_ud=spzeros(nudofs,nTdofs) ` and then asssemble it using my eledofs, it seems slow.

**Comment by KnutAM** [comment]
You could probably use the `sp = `[BlockSparsityPattern](https://ferrite-fem.github.io/Ferrite.jl/stable/reference/sparsity_pattern/#BlockSparsityPattern) (although the documentation is a bit scarce at the moment) in combination with [renumber!](https://ferrite-fem.github.io/Ferrite.jl/stable/reference/dofhandler/#Dof-renumbering):ing `FieldWise`. If you don't add nonzero values to the non-coupling matrices the overhead of having these should be negligible I think. For the assembly, then you should use the standard assembler obtained from `start_assemble(K, [f])`, where `K = allocate_matrix(sp)`. 

I haven't tried this functionality myself yet though.

**Reply by fredrikekre** [reply]
https://github.com/Ferrite-FEM/Ferrite.jl/discussions/1050 has some examples

**Reply by zyhsjtu** [reply]
Thanks for your suggestion, I will see this [#1050 ](url)
